### PR TITLE
[ci] add shutdowns option to createDatabase2 step

### PR DESCRIPTION
**Pull request by cseed**
shutdowns in createDatabase2 is a list of deployments to shut down before applying any migrations.

### CODE REVIEW

**Comment by cseed** [comment]
Note, this means migrations for batch will shut batch down, but batch won't be running to restart itself (!) so this will have to be done manually.  The UI will also be down, so we'll have to watch the database to verify the migration has been applied to know when to restart batch.

**Comment by jigold** [comment]
Is there an easy way to specify at the end once migrations have completed successfully to restart the deployments you cancelled once the migrations have succeeded? Maybe have k8s get the deployment yaml configuration before deleting each deployment and then reapply the yamls for each shutdown step in order.

I'm worried this PR will break all dev deploys with migrations in them as you can't just do `make -c batch deploy`

**Comment by jigold** [comment]
I'll look at the code carefully later.

**Comment by cseed** [comment]
> I'm worried this PR will break all dev deploys

The migration will bring down the service (batch, etc.) inside the dev namespace, but the dev deploy itself is running in the production batch, so there should be no issue.

The deploys will still happen as normal after the migrations, e.g. deploy_batch will run after batch_database.

File: ci/create_database.py [hunk 1]
```diff
@@ -126,6 +149,8 @@ def generate_token(size=12):
     current_version = row['version']
 
     if current_version + 1 == to_version:
+        shutdown()
```
Comment (danking) [inline ci/create_database.py]: await

Comment (cseed) [reply ci/create_database.py]: Argh, thanks.  I would give a lobe of my liver for a linter that checks asyncio errors.

**Comment by cseed** [comment]
> Is there an easy way to specify at the end once migrations have completed successfully to restart the deployments you cancelled once the migrations have succeeded?

This is an interesting idea.  I think it would effectively require merging the database and the deploy steps.  We can't re-deploy the existing config because that's the wrong (old) one.

File: ci/create_database.py (hunk 1 above)
Comment (danking) [reply ci/create_database.py]: seconded.

File: ci/create_database.py [hunk 2]
```diff
@@ -107,6 +110,26 @@ def generate_token(size=12):
     })
 
 
+did_shutdown = False
+
+
+async def shutdown():
```
Comment (jigold) [inline ci/create_database.py]: Why is did_shutdown a global variable and who is it used by?

Comment (cseed) [reply ci/create_database.py]: Shutdown is called inside the migrate function which is called once for each migration (in order).  Migration might already be applied, in which case the function does nothing.  We only shut down if we actually apply a migration, in which case shutdown is called.  But we only need to shutdown once, so did_shutdown records whether we've already shutdown.

**Comment by cseed** [comment]
I think I addressed all the comments.  Let me know if there is anything else.

**Pull request merged**